components/mpas-ocean/src/shared/mpas_ocn_transport_tests.F

```diff
@@ -0,0 +1,669 @@
+             else
+              ! Correlated Tracers test case
+              do k=1, nVertLevels
+                 debugTracers(1, k, iCell) = vs*xyztrig(lonCell(iCell), latCell(iCell))
+                 debugTracers(2, k, iCell) = vs*cosine_bells(lonCell(iCell), latCell(iCell))
+                 debugTracers(3, k, iCell) = correlation_fn(debugTracers(2, k, iCell))
+              enddo
+             endif
+          endif
+
+          do k = 1, nVertLevels
+             layerThickness(k, iCell) = 100.0_RKIND
+             restingThickness(k, iCell) = layerThickness(k, iCell)
+          end do
+
+          fCell(iCell) = 0.0_RKIND ! 0 Coriolis
+          bottomDepth(iCell) = 100.0_RKIND
+          minLevelCell(iCell) = 1
+          maxLevelCell(iCell) = nVertLevels
+       end do
+
+       do iEdge = 1, nEdgesSolve
+          fEdge(iEdge) = 0.0_RKIND
+       end do
+
+       do iVertex=1, nVerticesSolve
+          fVertex(iVertex) = 0.0_RKIND
+       end do
+
+       do iEdge = 1,nEdgesSolve
+          normalVelocity(:,iEdge) = 0
+       enddo
+
+       block_ptr => block_ptr % next
+    end do
+#ifdef MPAS_DEBUG
+    call mpas_log_write("exiting subroutine ocn_transport_test_init_setup", MPAS_LOG_OUT, .true. , flushNow=flush_log)
+#endif
+  end subroutine ocn_transport_test_init_setup
+
+!***********************************************************************
+!
+!  routine ocn_transport_test_init_validate
+!
+!> \brief Ensures that vertical levels = 3 for 2D tests
+!> \author Andrew Bradley
+!> \date   12/2/2020
+!> \details
+!>
+!
+!-----------------------------------------------------------------------
+subroutine ocn_transport_test_init_validate(cfgpl, pkgpl, iocontext, err)
+  type (mpas_pool_type), intent(inout) :: cfgpl, pkgpl
+  type (mpas_io_context_type), intent(inout) :: iocontext
+  integer, intent(out) :: err
+
+  integer, pointer :: config_vert_levels
+
+  err = 0
+
+  call mpas_pool_get_config(cfgpl, 'config_vert_levels', config_vert_levels)
+  config_vert_levels = 3
+end subroutine ocn_transport_test_init_validate
+
+!***********************************************************************
+!
+!  routine ocn_transport_test_velocity
+!
+!> \brief Sets edge normal velocities for prescribed flow fields
+!> \author Andrew Bradley
+!> \date   12/2/2020
+!> \details Sets edge normal velocities for prescribed flow fields
+!>
+!
+!-----------------------------------------------------------------------
+subroutine ocn_transport_test_velocity(meshPool, days_since_start, dt, normalVelocity)
+    type (mpas_pool_type), intent(in) :: meshPool
+    real(RKIND), intent(in) :: days_since_start, dt
+    real(RKIND), intent(out) :: normalVelocity(:,:)
+
+    real(RKIND), dimension(:), pointer :: latEdge, lonEdge, latVertex, lonVertex
+    integer, dimension(:,:), pointer :: verticesOnEdge
+    real(RKIND), pointer :: sphere_radius
+    real(RKIND) :: time_sec, nml(2)
+    integer, pointer :: nEdges
+    integer :: i
+
+    call flow_run()
```

Comment:
@pbosler can you move this to the init routine so the if statement is not checked every timestep?  Or is there a need to leave it here?  I do think we need a way to fully disable these tests for e3sm simulations.